source/core_mqtt.c

```diff
@@ -1913,13 +1914,16 @@ static MQTTStatus_t sendSubscribeWithoutCopy( MQTTContext_t * pContext,
 
     while( ( status == MQTTSuccess ) && ( subscriptionsSent < subscriptionCount ) )
```

Comment:
> Should this assert `assert( MQTT_SUB_UNSUB_MAX_VECTORS >= subscriptionStringVectorSlots )`; be changed to `assert( MQTT_SUB_UNSUB_MAX_VECTORS >= (subscriptionStringVectorSlots * subscriptionCount)`;

I don't think that the change you suggested is correct. It is asserting on maximum number of subscriptions it can send in one call to `sendMessageVector`. The function can split the remaining subscriptions in many smaller chunks without any problem.

> I don't see any logic that handles when a user tries to send a list of too many topics. If anything it looks like it will send as many as it can and then just return happy

That is how it is supposed to be. Unless the list becomes too large so as not to fit in a standard MQTT message, there is no restriction on how many topics filters can be sent in one packet. It is listed here in the spec: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718063

Reply:
Oh sorry I didn't realize that there was nested while loops when I was looking at this. I see now that it sends the message buffer as soon as IoVector cannot store another subscribe then continues.